### PR TITLE
Creating a library provider

### PR DESCRIPTION
**Pull request by jmauro**
- You can use attribute to define a profile as well as properties

This idea behind this PR is to be able to use the prodiver directly with properties as well as initial purpose of the cookbook + adding a library to handle the ini creation.

cc: @aboten, @criteo-cookbooks/sre-core

### CODE REVIEW

**Comment by kamaradclimber** [comment]
lgtm but you should do this PR upstream

**Comment by jmauro** [comment]
Of course wanted to validate with U guys first

File: resources/profile.rb [hunk 1]
```diff
@@ -1,2 +1,10 @@
 actions :create, :enable, :disable, :default
 default_action :create
+
+# man 'tuned.conf'
+# MAIN SECTION
+#   The main section is called "[main]"
+property :main, kind_of: Hash, default: {}
```
Comment (Annih) [inline resources/profile.rb]: your default value will be shared by every resource instance, you should at least freeze the empty hash

File: providers/profile.rb [hunk 2]
```diff
@@ -8,17 +8,22 @@ def whyrun_supported?
...
+    action [:enable, :start]
+  end
 
   # initialise empty attribute hash in case no attributes where specified
   # default [main] entry
-  node.default['tuned'] ||= {}
-  node.default['tuned']['profile'] ||= {}
-  node.default['tuned']['profile'][profile[:name]] ||= {}
-  node.default['tuned']['profile'][profile[:name]]['main'] ||= {}
+  node.default['tuned']['profile'][profile['name']] ||= {}
```
Comment (Annih) [inline providers/profile.rb]: I'm not sure you need to do that

File: providers/profile.rb [hunk 3]
```diff
@@ -67,6 +83,13 @@ def whyrun_supported?
 end
 
 action :default do
+  # Requirement
+  package 'tuned'
+
+  service 'tuned' do
+    action [:enable, :start]
+  end
```
Comment (Annih) [inline providers/profile.rb]: please factorize these lines, you are using them in all actions

**Comment by jmauro** [comment]
Review summiting the PR Upstream

**Pull request closed without merging**